### PR TITLE
fix(package_info_plus): Added build signature to package info data

### PR DESCRIPTION
**Pull request by acoutts**
## Description
Re-submission from https://github.com/flutter/plugins/pull/4013

This adds the android build signature to the package_info result. It returns an empty string on all other platforms.

For android this is useful when you need to get the SHA1 fingerprint which signed the current running app, for example when using platform-specific API keys for the Google Maps API where you must provide the signing certificate in a x-android-cert header.

## Related Issues
flutter/flutter#83996

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by mhadaily** [comment]
Can you please merge with `main`

**Comment by acoutts** [comment]
@mhadaily you want me to merge this into the main on my fork?

**Comment by mhadaily** [comment]
yes, please on your fork, update `main` branch with upstream and then merge your branch with `main` so you get the latest updates from our `main branch`

**Comment by acoutts** [comment]
Done.

File: packages/package_info_plus/package_info_plus/example/ios/Flutter/Flutter.podspec [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-Flutter provides an easy and productive way to build and deploy high-performance mobile apps for Android and iOS.
-                       DESC
-  s.homepage         = 'https://plus.fluttercommunity.dev'
+  s.homepage         = 'https://flutter.io'
   s.license          = { :type => 'MIT' }
-  s.author           = { 'Flutter Community' => 'authors@fluttercommunity.dev' }
+  s.author           = { 'Flutter Dev Team' => 'flutter-dev@googlegroups.com' }
   s.source           = { :git => 'https://github.com/flutter/engine', :tag => s.version.to_s }
   s.ios.deployment_target = '8.0'
-  s.vendored_frameworks = 'Flutter.framework'
```
Comment (mhadaily) [inline packages/package_info_plus/package_info_plus/example/ios/Flutter/Flutter.podspec:17]: We want to keep this as is.

**Review by mhadaily — CHANGES_REQUESTED** [review]

**Comment by mhadaily** [comment]
Thanks for the PR, 

this seems to be valuable for Android only. 

Let me ask @jpnurmi  and @miquelbeltran about their opinion. What do you guys think about this PR?

File: packages/package_info_plus/package_info_plus/example/ios/Flutter/Flutter.podspec (hunk 1 above)
Comment (acoutts) [reply packages/package_info_plus/package_info_plus/example/ios/Flutter/Flutter.podspec:17]: Sorry must be a merge conflict. I did not mean to edit this one.

**Comment by acoutts** [comment]
> Thanks for the PR,
> 
> this seems to be valuable for Android only.
> 
> Let me ask @jpnurmi and @miquelbeltran about their opinion. What do you guys think about this PR?

I think it's valuable because of the use case I laid out - when using Google Maps, Google Place, and other APIs which use platform-specific API keys you need to send the build signature as a header. There are probably other use cases to have the build signature at runtime.

It seems like there might be other data that is platform specific. What about taking the approach of Device Info where they return platform data objects specific to each platform?

**Comment by miquelbeltran** [comment]
1. Fine for adding the SHA signature, although I had never the need for it
2. the changes on the iOS code must be removed
3. we need to be sure that all dependencies between the platform interface and platform implementation projects are updated correctly

**Comment by acoutts** [comment]
Just removed the inadvertent iOS changes.

**Review by mhadaily — APPROVED** [review]
LGTM ,thank you

**Pull request merged**